PHPC-1399: Test MongoDB 4.2 in Evergreen

alcaeus opened the pull request:
In addition to adding MongoDB 4.2, I've also added PHP 7.3 as that was missing from the build matrix.

I've split the storage engine test to test MMAPv1 with MongoDB 4.0 and the InMemory and WiredTiger storage engines with MongoDB 4.2. I believe it's not necessary to also test the other storage engines with MongoDB 4.0, but I can add it if deemed necessary.

jmikola commented:
https://travis-ci.org/mongodb/mongo-php-driver/builds/565370854 looks like the correct Travis CI build, despite the cancelled build linked above.

Looks like there are some issues with the Evergreen configuration, though. I'm not sure if the `evergreen` binary has a validation command, but you could try `evergreen evaluate  .evergreen/config.yml` to see if it provides more helpful insight.

You may want to split out PHP 7.3 into a separate ticket/PR, as I believe that may require some action from the build team to add the version to the PHP driver's toolchain. See some of the tickets related to [PHPC-837](https://jira.mongodb.org/browse/PHPC-837) for more information on that, as they provide a record of the steps that Derick had to go through when we setup Evergreen last year.

alcaeus commented:
Yes, a tiny mistake (`mmapv1` vs. `mmap-v1`) messed this up. I've fixed that and extracted PHP 7.3 to https://jira.mongodb.org/browse/PHPC-1418. I'll take a look at that separately and then create any possible further tickets to add PHP 7.3 to the toolchain.
